modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/autocomplete/Autocomplete.js

```diff
@@ -12,346 +12,129 @@
 
-import {debouncePromise} from '../../utilities/debounce';
-import {AUTOCOMPLETE_VALUE_UPDATED} from '../../utilities/eventsDefinitions';
-import {useLiferayModule} from '../../utilities/hooks';
+import debounce from '../../utilities/debounce';
 import {getData, getValueFromItem} from '../../utilities/index';
 import {showErrorNotification} from '../../utilities/notifications';
 import InfiniteScroller from '../infinite_scroller/InfiniteScroller';
 
-function Autocomplete({onChange, onItemsUpdated, onValueUpdated, ...props}) {
-	const [query, setQuery] = useState(props.initialLabel || '');
-	const [initialised, setInitialised] = useState(
-		Boolean(props.customViewModuleUrl || props.customView)
-	);
-	const [active, setActive] = useState(false);
-	const [selectedItem, updateSelectedItem] = useState(props.initialValue);
-	const [items, updateItems] = useState(null);
+function Autocomplete({
+	apiUrl,
+	customView: CustomView,
+	disabled = false,
+	inputName,
+	inputPlaceholder = Liferay.Language.get('type-here'),
+	valueKey,
+	labelKey,
+	pageSize = 10,
+}) {
+	const [query, setQuery] = useState('');
+	const [selectedItem, setSelectedItem] = useState(null);
+	const [items, setItems] = useState(null);
 	const [loading, setLoading] = useState(false);
-	const [totalCount, updateTotalCount] = useState(null);
-	const [lastPage, updateLastPage] = useState(null);
-	const [page, updatePage] = useState(1);
-	const [pageSize, updatePageSize] = useState(props.pageSize);
-	const node = useRef();
-	const dropdownNode = useRef();
-	const inputNode = useRef();
-	const FetchedCustomView = useLiferayModule(props.customViewModuleUrl);
+	const [totalCount, setTotalCount] = useState(null);
+	const [lastPage, setLastPage] = useState(null);
+	const [page, setPage] = useState(1);
+	const [internalPageSize, setInternalPageSize] = useState(pageSize);
 	const isMounted = useIsMounted();
-	const [debouncedGetItems, updateDebouncedGetItems] = useState(null);
-
-	useEffect(() => {
-		updateDebouncedGetItems(() =>
-			debouncePromise(getData, props.fetchDataDebounce)
-		);
-	}, [props.fetchDataDebounce]);
-
-	const currentValue = selectedItem
-		? getValueFromItem(selectedItem, props.itemsKey)
-		: null;
-	const currentLabel = selectedItem
-		? getValueFromItem(selectedItem, props.itemsLabel)
-		: null;
-
-	const CustomView = props.customView || FetchedCustomView;
-
-	useEffect(() => {
-		if (items && items.length === 1 && props.autofill) {
-			const firstItem = items[0];
-			updateSelectedItem(firstItem);
-		}
-	}, [items, props.autofill, props.itemsKey, props.itemsLabel]);
-
-	useEffect(() => {
-		const value =
-			selectedItem && getValueFromItem(selectedItem, props.itemsKey);
-
-		if (props.id) {
-			Liferay.fire(AUTOCOMPLETE_VALUE_UPDATED, {
-				id: props.id,
-				itemData: selectedItem,
-				value,
-			});
-		}
-
-		if (onValueUpdated) {
-			onValueUpdated(value, selectedItem);
-		}
 
-		if (onChange) {
-			onChange({target: {value}});
-		}
```

Comment:
These props are never used. Meaning <Autocomplete is never used and provided these props.

I am starting to think this component may have just been originally copied from some other place and a bunch of this logic got included. Or perhaps it used to be necessary and was never removed.

Reply:
Also `onValueUpdated`, was created for a purpose (https://issues.liferay.com/browse/COMMERCE-4265) and it's being used here:

```
6 results - 5 files

modules/apps/commerce/commerce-catalog-web/src/main/resources/META-INF/resources/commerce_catalog/details.jsp

modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/commerce_channel/general.jsp

modules/apps/commerce/commerce-pricing-web/src/main/resources/META-INF/resources/commerce_price_lists/commerce_price_list/details.jsp

modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/definition/add_cp_definition.jsp

modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/definition/duplicate_cp_definition.jsp
```

Reply:
@FabioDiegoMastrorilli thanks for pointing those instances out, I hadn't seen them before. I will check it out.